src/index.ts

```diff
@@ -154,50 +146,65 @@ export default class QuickbooksConnector extends BaseHttpConnector<
     return false
   }
 
-  private async refreshTokenIfNeeded() {
-    const dbToken = await this.getQBToken()
-    if (!dbToken) {
-      console.error(
-        `refreshTokenIfNeeded, stored token not found for ${this.configOptions.realmId}`,
-      )
-      return
-    }
-    const expiry = Number(dbToken.token.createdAt) + Number(dbToken.token.expires_in) * 1000
-    const needRefresh = expiry - 120 * 1000 < Date.now() // 2 minutes before token is expired
-
-    if (needRefresh) {
-      console.log('Before refreshing')
-      try {
-        this.oauthClient.setToken(dbToken.token)
-        const authResponse = await this.oauthClient.refresh()
-        await this.storeTokenAndSetClient(authResponse)
-        console.log('Token is refreshed')
-      } catch (e: any) {
-        console.error('Refresh Token Error:', e)
-        console.error(e.intuit_tid)
+  private async refreshTokenWhenNeeded() {
+      const dbToken = await this.getQBToken()
+      if(!dbToken) {
+        this.refreshIsRunning = false
+        return
       }
+
+      const expiry = this.timeToRefreshToken(dbToken)
```

Comment:
`expiry` is only used once - so maybe we could drop the declaration and just use `this.timeToRefreshToken(dbToken)` at the end of the `setTimeout` call